src/main/java/org/apache/commons/imaging/formats/tiff/datareaders/ImageDataReader.java

```diff
@@ -518,8 +519,71 @@ protected void applyPredictorToBlock(final int width, final int height, final in
         return samples;
     }
 
+
+    protected int[] unpackIntSamples(
+        final int width,
+        final int height,
+        final int scansize,
+        final byte[] bytes,
+        final int predictor,
+        final int bitsPerSample,
+        final ByteOrder byteOrder)
+        throws ImageReadException {
+        final int bytesPerSample = bitsPerSample / 8;
+        final int nBytes = bytesPerSample * scansize * height;
+        final int length = bytes.length < nBytes ? nBytes / scansize : height;
+
+        final int[] samples = new int[scansize * height];
+        // At this time, Commons Imaging only supports two-byte
+        // two's complement short integers.  It is assumed that
+        // the calling module already checked the arguments for
+        // compliance, so this method simply assumes that they are correct.
+
+        // The logic that follows is simplified by the fact that
+        // the existing API only supports two-byte signed integers.
+        boolean useDifferencing
+                = predictor == TiffTagConstants.PREDICTOR_VALUE_HORIZONTAL_DIFFERENCING;
+
+        for (int i = 0; i < length; i++) {
+            int index = i * scansize;
+            int offset = index * 2;
+            if (byteOrder == ByteOrder.LITTLE_ENDIAN) {
+                for (int j = 0; j < width; j++) {
+                    int a = bytes[offset++] & 0xff;
+                    int b = bytes[offset++] & 0xff;
+                    int c = (b<<8)|a;
+                    if((c&0x8000)!=0){
+                        c|=0xffff0000;
+                    }
+                    samples[index + j] = c;
+                }
+            } else {
+                for (int j = 0; j < width; j++) {
+                    int a = bytes[offset++] & 0xff;
+                    int b = bytes[offset++] & 0xff;
+                      int c = (b<<8)|a;
+                    if((c&0x8000)!=0){
+                        c|=0xffff0000;
```

Comment:
@gwlucastrig you calculate `c` here, but never effectively uses it I think?

Reply:
Good catch.  This code is wrong.  This logic is supposed to handle the case where the sign bit for the short integer is set. Because the bit shifting stores the source bits in a four-byte integer, the high-order two bytes are clear.  So the idea is to make sure that the sign bits are set.

The unit tests didn't catch this because they don't populate the sample with values in the negative range.  I'll have to fix that.

Of course, this glitch could probably have been done better just by relying on Java to to the casting.  I will review the Java rules for casting an int to a short when the int is in the range > 32767 (0x8000)

Reply:
done.  Unit test modified to verify negative samples.